src/system_metrics/service_mongoose_system_metrics.erl

```diff
@@ -35,25 +40,63 @@ start_link(Args) ->
 init(Args) ->
     InitialReport = proplists:get_value(initial_report, Args, ?DEFAULT_INITIAL_REPORT),
     ReportAfter = proplists:get_value(report_after, Args, ?DEFAULT_REPORT_AFTER),
-    load_client_id(),
-    erlang:send_after(InitialReport, self(), spawn_gatherer),
-    {ok, ReportAfter}.
+    erlang:send_after(InitialReport, self(), spawn_collector),
+    {ok, #system_metrics_state{report_after = ReportAfter}}.
     
-handle_info(spawn_gatherer, ReportAfter) ->
-    spawn(mongoose_system_metrics_gatherer, gather, []),
-    erlang:send_after(ReportAfter, self(), spawn_gatherer),
-    {noreply, ReportAfter};
+handle_info(spawn_collector, #system_metrics_state{report_after = ReportAfter, collector_monitor = none} = State) ->
+    case get_client_id() of
+        {error, no_client_id} -> {stop, no_client_id, State};
+        {ok, ClientId} ->
+            {_Pid, Monitor} = spawn_monitor(mongoose_system_metrics_collector, collect, [ClientId]),
+            erlang:send_after(ReportAfter, self(), spawn_collector),
+            {noreply, State#system_metrics_state{collector_monitor = Monitor}}
+    end;
+handle_info({'DOWN', CollectorMonitor, _, _, _}, #system_metrics_state{collector_monitor = CollectorMonitor} = State) ->
+    {noreply, State#system_metrics_state{collector_monitor = none}};
 handle_info(_Message, _State) ->
     ok.
 
+
 % %%-----------------------------------------
 % %% Helpers
 % %%-----------------------------------------
 
-load_client_id() ->
-    ClientId = uuid:uuid_to_string(uuid:get_v4()),
-    persistent_term:put(ga_client_id, ClientId).
-%TODO: save client id to file and if file exists, read from file
+% trying to get client ID 20 times, because it seems fine
```

Comment:
I think it'd be beneficial for future maintainers of the code to know why exactly these retires are needed. Could you describe it shortly in the comment?